fitbenchmarking/fitbenchmarking
# 734 access internal jacobians

#### Description of Work

Fixes #734 by adding a new Jacobian option, ``default``, which uses the Jacobian supplied by the solver in the case that no Jacobian is supplied.  This replaces the `<solver>-no-jac` functionality that was present to access this before, and those solvers have been removed.

I've also updated some of the tests to make them easier to update/troubleshoot.


#### Testing Instructions

1. Run tests with `jac_method = default`, and check results are as expected
2. Check results are the same as for `scipy_no_jac`
3. Read through the docs and check they are clear

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
